src/app/engine/audio-backend.service.ts

```diff
@@ -110,13 +110,27 @@ export class AudioBackendService {
   }
 
   private loadBank(url: string) {
+    // on Safari we need to use callbacks using decodeAudioData method.
     this.http
+      .get(url, { responseType: 'arraybuffer' })
+      .toPromise()
+      .then( (result: any) => {
+         this.context.decodeAudioData(result, (buffer) => {
```

Comment:
I think it will be better to wrap this in a promise, e.g.
`new Promise(resolve => this.context.decodeAudioData(result, resolve))`
this will return a promise so we don't have to change the rest of the code.

Reply:
Nice! I'll modify it this week...

Reply:
new PR sent, I just adapted the code, I hope I understood you correctly